client/mysqlbinlog.cc

```diff
@@ -3687,11 +3687,23 @@ int main(int argc, char** argv)
     start_position= BIN_LOG_HEADER_SIZE;
   }
 
+  /*
+    Emit a warning in the event that we finished processing input
+    before reaching the boundary indicated by --stop-datetime.
+  */
   if (stop_datetime != MY_TIME_T_MAX &&
       stop_datetime > last_processed_datetime)
     warning("Did not reach stop datetime '%s' before end of input",
             stop_datetime_str);
 
+  /*
+    Emit warning(s) (in Gtid_event_filter::verify_completed_state())
+    in the event that we finished processing input
+    before reaching all boundaries indicated by --stop-position GTID(s).
+  */
+  if (position_gtid_filter)
+    position_gtid_filter->verify_final_state();
```

Comment:
When implemented,
```suggestion
  if (gtid_event_filter)
    gtid_event_filter->verify_final_state();
```
will additionally warn if we finished processing before
* reaching all `--start-position` GTID(s), and
* touching all `--ignore-domain`/`server-ids`.